### DISCUSSION TITLE
Is it necessary to have two command lines for wgnuplot and gnuplot?

### DISCUSSION THREAD

**Discussion by jalvesz** [kookma/ogpf#46] [Q&A]
Hi,
I'm testing this repo under windows and linux. I found out that after installing gnuplot and running the examples I had an error here:
```Fortran
if (get_os_type() == 1) then
            call execute_command_line ('wgnuplot -persist ' // this%txtfilename)
```
As wgnuplot is not recognized, but actually gnuplot is. So basically, the same command line works for both OSs.

I just wanted to know if there is a specific reason for keeping both?

Thanks,

**Comment by kookma** [comment]
The `wgnuplot` works under Windows. Do you run the ogpf under Windows?

**Reply by jalvesz** [reply]
Yes, I run under windows, after installing gnuplot, the command with "wgnuplot" was not working but "gnuplot" did launch. I did manage after to get the wgnuplot command working by creating a symbolic link with:
```bash
mklink wgnuplot "C:\Program Files\gnuplot\bin\wgnuplot.exe" 
```
Maybe is just an install issue, in practice everything works. Was just wondering if there is a specific reason for distinguishing the commands?

BTW, thanks for this very nice package!! :)

**Reply by kookma** [reply]
Open a terminal and run gnuplot and wgnuplot separately.
See the difference between two! The second uses a nice GUI, while the first is a command-line based program.

**Comment by kookma** [comment]
This answer is generated by Bing AI and it is correct.

The primary difference between **wgnuplot** and **gnuplot** is that **wgnuplot** uses the wxWidgets library for its GUI, while **gnuplot** is a command-line driven program ¹. 

**Gnuplot** is a versatile function and data plotting program that can generate two- and three-dimensional plots of functions, data, and data fits ¹. It is available on various platforms such as Linux, OS/2, MS Windows, OSX, VMS, and more ⁵. 

**Wgnuplot** is a fork of **gnuplot** that uses the wxWidgets library for its GUI ⁴. The actual drawing is done via cairo, a 2D graphics library, and pango, a library for laying out and rendering text ⁴. This terminal should work on all three OS ⁴.

I hope this helps!

Source: Conversation with Bing, [1](http://gnuplot.info/docs_4.2/gnuplot.html)2/[1](http://gnuplot.info/docs_4.2/gnuplot.html)5/[2](http://www.gnuplot.info/)02[3](https://stackoverflow.com/questions/22784344/gnuplot-what-is-difference-between-wxt-and-windows-terminal)
(1) Contents. http://gnuplot.info/docs_4.2/gnuplot.html.
(2) gnuplot homepage. http://www.gnuplot.info/.
([3](https://stackoverflow.com/questions/22784344/gnuplot-what-is-difference-between-wxt-and-windows-terminal)) GNUPLOT: what is difference between wxt and windows terminal?. https://stackoverflow.com/questions/22784[3](https://stackoverflow.com/questions/22784344/gnuplot-what-is-difference-between-wxt-and-windows-terminal)44/gnuplot-what-is-difference-between-wxt-and-windows-terminal.
(4) MATLAB vs GnuPlot: Which is Better? (2021) - Appmus. https://appmus.com/vs/matlab-vs-gnuplot.
(5) GnuPlot vs MATLAB: Which is Better? (2021) - Appmus. https://appmus.com/vs/gnuplot-vs-matlab.
(6) Best practice for plots (pgfplots, gnuplot, etc.). https://tex.stackexchange.com/questions/555218/best-practice-for-plots-pgfplots-gnuplot-etc.
